tekstowo.py

```diff
@@ -0,0 +1,464 @@
+# decide how to save progress: either txt with specific formatting or a json file - DONE
+# progress bar?
+
+# Function collection
+def generate_timestamp() -> str:
+    """
+    Returns current system time.
+    """
+    # Get the current time
+    current_datetime = datetime.now()
+
+    # Format the date and time as a string
+    timestamp = current_datetime.strftime("%Y-%m-%d, %H:%M:%S")
+
+    return timestamp
+
+def processing_time(start_timestamp: datetime, end_timestamp: datetime) -> Tuple[int, int, int, int]:
+    """
+    Counts the time that has passed performing a given task.
+    """
+    processing_time = end_timestamp - start_timestamp
+    total_seconds = processing_time.total_seconds()
+
+    days = int(total_seconds // (3600 * 24))
+    hours = int((total_seconds // 3600) % 24)
+    minutes = int((total_seconds % 3600) // 60)
+    seconds = int(total_seconds % 60)
+
+    return days, hours, minutes, seconds
+
+def get_max_page_number(url: str) -> int:
+    """
+    Get the highest page number per given letter.
+    Used in cooperation with 'create_lut_pagination' 
+    and 'pages_per_letter' method.
+    """
+
+    max_page = 0
+
+    try:
+        req = requests.get(url, timeout=60)
+        req.raise_for_status()
+    
+    except requests.exceptions.RequestException as e:
+        print(f"An error occured: {e}")
+        return max_page
+
+    if req.ok:
+        soup = BeautifulSoup(req.content, 'lxml')
+
+        if soup:
+            for page in soup.find_all(class_='page-link'):
+                if page.text.isnumeric() and int(page.text) > max_page:
+                    max_page = int(page.text)
+
+    return max_page
+
+def create_lut_pagination() -> dict:
+    """
+    Creates a look-up table for each letter containing max page number
+    per letter in 'alphabet' variable. By default these are all26 letters 
+    of the alhpabet with an addition of 'pozostale' entry, which is a 
+    domain specific requirement.
+    """
+
+    alphabet = list(string.ascii_uppercase) + ['pozostale']
+
+    lut_pages = {}
+
+    for letter in alphabet:
+        time.sleep(1)
+        url = f"https://www.tekstowo.pl/artysci_na,{letter}.html"
+
+        try:
+            lut_pages[letter] = get_max_page_number(url)
+            print(f"Letter {letter} has {lut_pages[letter]} pages of artists.")
+
+        except Exception as e:
+            print(f"An error has occured for letter {letter}: {e}")
+   
+    return lut_pages
+
+def pages_per_letter(ARTIST_LETTER: str) -> dict:
+    """
+    Returns the value of the last page containing the songs
+    per artist.
+    """
+
+    letter_max_page = 0
+
+    if len(ARTIST_LETTER) < 2:
+        ARTIST_LETTER = ARTIST_LETTER.upper()
+    
+    url = f"https://www.tekstowo.pl/artysci_na,{ARTIST_LETTER}.html"
+
+    try:
+        letter_max_page = get_max_page_number(url)
+        print(f"Letter {ARTIST_LETTER} has {letter_max_page} pages of artists.")
+
```

Comment:
A single space isn't need in the try-except block of code as it is one, continuous 'block entity'.

Reply:
Changed to no space in try-except block.